File: Quorum3/Library/Standard/Plugins/javascript/Libraries/System/DateTime.js
function plugins_quorum_Libraries_System_DateTime_() {
    this.date = new Date();
    
    this.GetYear = function () {
        return date.getFullYear();
    };  

    this.GetMonth = function () {
        return date.getMonth();
    };  
    
    this.GetDayOfMonth = function () {
        return date.getDate();
    };  
    
    this.GetDayOfWeek = function () {
        return date.getDay();
    };  
    
    this.GetHour = function () {
        return date.getHour();
    };  
    
    this.GetMinute = function () {
        return date.getMinute();
    };  
    
    this.GetSecond = function () {
        return date.getSecond();
    };  
    
    this.GetTimeZone = function () {
        return date.getTimezoneOffset();
    };  
    
    this.IsDaylightSavings = function () {
        return date.getTimezoneOffset();
    };  
    
    this.GetEpochTime = function () {
        return Date.now().getUTCMilliseconds();
    };  
    
    this.SetEpochTime$quorum_number = function (value) {
        date.setUTCMilliseconds(value);
    };  
    
    this.SetTimeZone$quorum_integer = function (value) {
        //not supported in JS
    };  
}

File: Plugins/SystemPlugin/src/plugins/quorum/Libraries/System/DateTime.java
package plugins.quorum.Libraries.System;

/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * The default implementation of the date time plugin for the compiler.
 * 
 * @author Andreas Stefik
 */
public class DateTime {
    public java.lang.Object me_ = null;
    private QuorumDateTime dateTime = new QuorumDateTime();
    
    public int GetYear() {
        return dateTime.GetYear();
    }
    public int GetMonth() {
        return dateTime.GetMonth();
    }
    public int GetDayOfMonth() {         
        return dateTime.GetDayOfMonth();     
    }
    public int GetDayOfWeek() {         
        return dateTime.GetDayOfWeek();     
    }
    public int GetHour() {         
        return dateTime.GetHour();     
    }
    public int GetMinute() {         
        return dateTime.GetMinute();     
    }
    public int GetSecond() {         
        return dateTime.GetSecond();     
    }
    public int GetTimeZone() {         
        return dateTime.GetTimeZone();     
    }
    public boolean IsDaylightSavings() {         
        return dateTime.IsDaylightSavings();     
    }
    public double GetEpochTime() {         
        return dateTime.GetEpochTime();     
    }
    public void SetEpochTime(double epochTime) {
        dateTime.SetEpochTime((long)epochTime);
    }
    public void SetTimeZone(int timeZoneOffset) {
        dateTime.SetTimeZone(timeZoneOffset);
    }
}
